### PR TITLE
Fixes and improvements in cvem monitoring

### PR DESCRIPTION
**Pull request by feldsam**
Signed-off-by: Kristián Feldsam feldsam@gmail.com

I tested it on production environment, it's working fine.

Look at comments in commit

### CODE REVIEW

File: cvem/connectors/one/OpenNebula.py [hunk 1]
```diff
@@ -148,10 +148,10 @@ def get_vm_list():
 			res = []
 			for vm in res_vm.VM:
 				host = HostInfo(int(vm.HISTORY_RECORDS.HISTORY[0].HID), vm.HISTORY_RECORDS.HISTORY[0].HOSTNAME)
-				new_vm = VirtualMachineInfo(int(vm.ID), host, vm.TEMPLATE.MEMORY * 1024, vm)
+				new_vm = VirtualMachineInfo(int(vm.ID), host, int(vm.MAXMEMORY), vm)
```
Comment (micafer) [inline cvem/connectors/one/OpenNebula.py]: The VM.allocated_memory is the amount of memory originally allocated to this VM. So this value is the MEMORY field in the original TEMPLATE. Why do you think that is better to use MAXMEMORY?

Comment (feldsam) [inline cvem/connectors/one/OpenNebula.py]: Hi, it is outdated diff now, I revert it back...

File: cvem/connectors/one/OpenNebula.py [hunk 2]
```diff
@@ -148,10 +148,10 @@ def get_vm_list():
 			res = []
 			for vm in res_vm.VM:
 				host = HostInfo(int(vm.HISTORY_RECORDS.HISTORY[0].HID), vm.HISTORY_RECORDS.HISTORY[0].HOSTNAME)
-				new_vm = VirtualMachineInfo(int(vm.ID), host, vm.TEMPLATE.MEMORY * 1024, vm)
+				new_vm = VirtualMachineInfo(int(vm.ID), host, int(vm.TEMPLATE.MEMORY) * 1024, vm)
 				new_vm.user_id = vm.UID
 				if vm.USER_TEMPLATE.MEM_TOTAL:
-					new_vm.set_memory_values(int(vm.USER_TEMPLATE.MEM_TOTAL_REAL),
+					new_vm.set_memory_values(int(vm.REALMEMORY),
```
Comment (micafer) [inline cvem/connectors/one/OpenNebula.py:154]: The real_memory is the amount of memory assigned to the VM (in most of the cases is the same as allocated_memory). Here you are using REALMEMORY defined as the current memory usage.

Comment (feldsam) [inline cvem/connectors/one/OpenNebula.py:154]: Hi, some cut from CPMInfo.py

```
    self.real_memory =  None
    """ Real memory assigned to the VM """
    self.total_memory =  None
    """ Total memory reported by the S.O. (less than real_memory) """
    self.free_memory =  None
    """ Free memory of the VM """
    self.allocated_memory = allocated_memory
    """ Amount of memory originally allocated by the CMP """
```

So real memory have to be equal to virsh dominfo "Used memory ", that it is that memory, which we are set by virsh setmem command. So this is absolutely right.

Your onegate_published script reports vm.USER_TEMPLATE.MEM_TOTAL_REAL as value which is almost some as allocated_memory

Comment (micafer) [inline cvem/connectors/one/OpenNebula.py:154]: The VM.real_memory value is used in the monitor to calculate the difference between the real_memory and the total_memory reported by the O.S. So this memory must be the value that in most of the cases is the same as allocated_memory and allways must be higher that total_memory (as stated in the comments (Total memory reported by the S.O. (less than real_memory) ) and not the current used memory.

Comment (feldsam) [inline cvem/connectors/one/OpenNebula.py:154]: I know about that difference. Differenc eis between virsh Used memory and Total memory reported by OS. In all case real_memory is higher that total_memory, but it is noting to do with allocated memory which is original maximum memory which can be used by vm.

Comment (feldsam) [inline cvem/connectors/one/OpenNebula.py:154]: If you are open to skype call, them we can look on code and explain it. hmm? My skype tio289

File: cvem/connectors/one/OpenNebula.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ class DISK(XMLObject):
 		values = ['CLONE','READONLY','SAVE','SOURCE','TARGET' ]
 
 class TEMPLATE(XMLObject):
-		values = [ 'CPU', 'MEMORY', 'NAME', 'RANK', 'REQUIREMENTS', 'VMID', 'VCPU', 'PACKEDMEMORY', 'MAXMEMORY' ]
+		values = [ 'CPU', 'MEMORY', 'NAME', 'RANK', 'REQUIREMENTS', 'VMID', 'VCPU' ]
```
Comment (micafer) [inline cvem/connectors/one/OpenNebula.py:38]: As PACKEDMEMORY and MAXMEMORY are located in the TEMPLATE (at least in versions <= 4.12) is better to maintain these items in this list to make it work in both versions. I will modify it.

**Pull request merged**